### PR TITLE
UF-4725: Added optional validation of Role

### PR DESCRIPTION
**Pull request by MickeWallner**

### CODE REVIEW

File: src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidRoleConstraintValidator.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+	@Override
+	public boolean isValid(final String value, final ConstraintValidatorContext context) {
+		final var namespace = getPathVariable(PATHVARIABLE_NAMESPACE);
+		final var municipalityId = getPathVariable(PATHVARIABLE_MUNICIPALITY_ID);
+		return  !metadataService.isValidated(namespace, municipalityId, EntityType.ROLE) ||
+			isValid(value, getRoles(namespace, municipalityId), context);
+	}
+
+	private List<String> getRoles(String namespace, String municipalityId) {
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidRoleConstraintValidator.java:28]: I'm just curious:

The roleRepository has a existsByNamespaceAndMunicipalityIdAndName().
Wouldn't it be better to call this method (through the service-layer) instad of looping through a list of roles?

Is there any advantages with the current (looping) solution?

Comment (MickeWallner) [reply src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidRoleConstraintValidator.java:28]: I can not see any advantages with current solution and I can not see any reason for not changing handling of all types as you propose.

Comment (MickeWallner) [reply src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidRoleConstraintValidator.java:28]: When I look closer I find that the disadvantage is that the errorMessage will not show valid tags in your proposal.

Comment (foobar-1337) [reply src/main/java/se/sundsvall/supportmanagement/api/validation/impl/ValidRoleConstraintValidator.java:28]: ok

**Review by foobar-1337 — APPROVED** [review]

**Pull request merged**